docs/multiple_inputs.md

```diff
@@ -0,0 +1,166 @@
+metadata:
+  aus: "data/example_metadata_aus.tsv"
+  worldwide: "data/example_metadata_worldwide.tsv"
+```
+
+# Creating combined metadata
+
+The different provided metadata files (`aus` and `worldwide`, defined above) are combined during the pipeline.
+The combined metadata file includes all columns present: the `worldwide` metadata contains many more columns than the `aus` metadata does, so the latter samples will have a number of empty values.
+
+In the case of conflicts, the order of the entries in the YAML matters, with the last value being used.
+
+Finally, extra columns will be added for each input (e.g. `aus` and `worldwide`), with values `"yes"` or `"no"`, representing which samples are contained in each set of sequences.
+We are going to use this to our advantage, by adding a coloring to highlight the source of sequences in auspice via `my_profiles/example_multiple_inputs/my_auspice_config.json`:
+
+```json
+"colorings": [
+  {
+    "key": "aus",
+    "title": "Source: Australia",
+    "type": "boolean"
+  }
+],
+"display_defaults": {
+  "color_by": "aus"
+}
+```
+
+# (Pre-) Filtering  input-specific parameters
+
+The parameters used for filtering steps are typically defined by the "filter" dict in the `builds.yaml`, with sensible defaults provided (by `defaults/parameters.yaml`).
+For multiple inputs, we can overwrite these for each input.
+
+In this tutorial, we wish to make sure we include all the Australian samples, even if they may be partial genomes etc
+
+per-input level, such as the example tutorial does for `input1` (the North American genomes).
+
+```yaml
+# my_profiles/example_multiple_inputs/builds.yaml
+filter:
+  aus:
+    min_length: 5000 # Allow shorter genomes. Parameter used in the prefilter & filter rules
+    exclude_where: country=Canada # Would remove all Canadian sequences (there aren't any!)
+    min_date: "2020-02-01" # used by the filter rule. Will remove all sequences from Jan 2020
+    exclude_ambiguous_dates_by: year # used by the filter rule.
+    skip_diagnostics: True # skip diagnostics (which can remove genomes) for this input
+```
+
+# Subsampling parameters
+
+For subsampling, we utilise the fact that the combined metadata has additional columns to represent the input source: `aus` and `worldwide`.
+This allows us to have per-input subsampling steps by restricting the step to sequences from an individual input (or inputs).
+
+
+In this example, we want to include _all_ of the samples from `aus` (i.e. all Australian genomes) and then create a contextual subsampling of the genomes from `worldwide` based on genetic distance from the `aus` sample.
+
+```yaml
+builds:
+  multiple-inputs:
+    subsampling_scheme: custom-scheme # use a custom subsampling scheme defined below
+
+subsampling:
+  custom-scheme:
+    # Use metadata key to include ALL from `input1`
+    allFromAus:
+      exclude: "--exclude-where 'aus=no'" # subset to sequences from input `aus`
```

Comment:
I might have expected a single new column of `input-source` with fields of `aus` vs `worldwide` rather than the "one-hot" encoding that is used here.

Reply:
I went with one-hot as I often have the use case where the sequence sets are overlapping -- e.g. some of my "private" sequences are _also_ in GISAID / NCBI, and I couldn't think of how to nicely express this in a single variable whilst facilitating subsampling based on this variable...